ethereum_history_api/oracles/src/noir/oracles/server.test.ts

```diff
@@ -0,0 +1,64 @@
+import { describe, it, expect } from 'vitest';
+import { withOracleServer } from './server.js';
+import { createMockClient } from '../../ethereum/mockClient.js';
+
+const JSON_RPC_PAYLOAD = {
+  jsonrpc: '2.0',
+  method: 'get_header',
+  params: [{ Single: { inner: 'd895ce' } }],
+  id: 1
+};
+
+const GET_HEADER_POST_DATA = {
+  method: 'POST',
+  headers: { 'Content-Type': 'application/json' },
+  body: JSON.stringify(JSON_RPC_PAYLOAD)
+};
+
+const ORACLE_SERVER_PORT = 5556;
+
+type LocalhostOracleStatus = 'OK' | 'NOT_AVAILABLE';
+
+async function localhostOracleStatus() {
+  let statusText: LocalhostOracleStatus;
+  try {
+    const response = await fetch(`http://localhost:${ORACLE_SERVER_PORT}`, GET_HEADER_POST_DATA);
+    statusText = response.status == 200 ? 'OK' : 'NOT_AVAILABLE';
+  } catch (e) {
+    statusText = 'NOT_AVAILABLE';
+  }
+  return statusText;
+}
+
+describe('oracle server', async () => {
+  const mockClient = await createMockClient('./fixtures/mockClientData.json');
+
+  it('withOracleServer should start server', async () => {
+    const oracleStatus = await withOracleServer(
+      async () => {
+        return await localhostOracleStatus();
```

Comment:
How about inline?
```ts
const {status} = await fetch(SERVER_URI, GET_HEADER_POST_DATA);
expect(status).toEq(200)
```

Reply:
Introduced `function expectServerUp(serverUrl: string)` with the above code.